relay/transformer/transformer.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transformer
+
+import (
+	"github.com/pingcap/parser"
+	"github.com/siddontang/go-mysql/mysql"
+	"github.com/siddontang/go-mysql/replication"
+)
+
+var (
+	artificialFlag = uint16(0x0020) // LOG_EVENT_ARTIFICIAL_F
+)
+
+// Result represents a transform result.
+type Result struct {
+	Ignore      bool          // whether the event should be ignored
+	LogPos      uint32        // binlog event's End_log_pos or Position in RotateEvent
+	NextLogName string        // next binlog filename, only valid for RotateEvent
+	GTIDSet     mysql.GTIDSet // GTIDSet got from QueryEvent and XIDEvent when RawModeEnabled not true
```

Comment:
it's gtid or gtidset?

Reply:
👍 It's gtid set. It's GTID in one binlog event, but _go-mysql_ returns a GTID set (including the previous/passed in GTID set)